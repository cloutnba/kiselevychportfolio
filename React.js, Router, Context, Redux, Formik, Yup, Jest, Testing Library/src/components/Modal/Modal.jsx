import { useEffect, useRef} from "react";
import PropTypes from 'prop-types';
import './Modal.scss';
import Button from "../Button";


function Modal({text,onClickHandler, closeModal}) {
    const wrapperRef = useRef();

    useEffect(() => {
        document.addEventListener("mousedown", handleClickOutside);

        return (() => {
            document.removeEventListener("mousedown", handleClickOutside);
        })
    })

    const handleClickOutside = (event) => {
        if (wrapperRef && !wrapperRef.current.contains(event.target)) {
            closeModal();
        }
    }

    return(
        <div className="modal-wrapper">
            <div className="modal" ref={wrapperRef}>
                <div className="modal-box">
                    <Button className="modal-close" onClickHandler={closeModal}>
                        <svg viewBox="0 0 16 16" width="16" height="16">
                            <path
                                d="m9.3 8 6.1-6.1c.4-.4.4-.9 0-1.3s-.9-.4-1.3 0L8 6.7 1.9.6C1.6.3 1 .3.6.6c-.3.4-.3 1 0 1.3L6.7 8 .6 14.1c-.4.4-.3.9 0 1.3l.1.1c.4.3.9.2 1.2-.1L8 9.3l6.1 6.1c.4.4.9.4 1.3 0s.4-.9 0-1.3L9.3 8z"/>
                        </svg>
                    </Button>
                    <div className="modal-content">
                        <div className="modal-text">
                            {text}
                        </div>
                    </div>
                    <div className="modal-footer">
                        <div className="button-wrapper">
                            <Button className="btn" onClickHandler={onClickHandler}>ОК</Button>
                            <Button className="btn" onClickHandler={closeModal}>НІ</Button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

Modal.propTypes = {
    text: PropTypes.string,
    handlerAddItem: PropTypes.func,
    closeModal: PropTypes.func,
}

export default Modal;
