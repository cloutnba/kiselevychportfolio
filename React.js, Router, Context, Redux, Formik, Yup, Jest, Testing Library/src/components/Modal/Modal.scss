.modal-wrapper {
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	z-index: 100;
	background-color: rgba(0,0,0,0.4);
}

.modal {
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%,-50%);
	max-width: 560px;
	width: 100%;
	padding: 40px;
	box-sizing: border-box;
	background-color: #fff;
	border-radius: 4px;
	height: 150px;

	.modal-close {
		position: absolute;
		right: 16px;
		top: 16px;
		display: inline-flex; vertical-align: middle; text-align: center; box-sizing: border-box;
		align-items: center; justify-content: center;
		text-decoration: none !important;
		outline: none;
		-webkit-appearance: none;
		border: none;
		cursor: pointer;
		user-select: none;
		background-color: transparent;
	}


	h4 {
		margin: 0;
	}

	.modal-content {}

	.modal-footer {
		margin-top: 24px;
	}

	.button-wrapper {
		display: flex;
		justify-content: space-between;
		align-items: center;
	}
}

.btn {
	position: relative;
	display: inline-flex; vertical-align: middle; text-align: center; box-sizing: border-box;
	align-items: center; justify-content: center;
	text-decoration: none !important;
	outline: none;
	-webkit-appearance: none;
	border: none;
	cursor: pointer;
	user-select: none;
	background-color: #6b78d6;
	font-family: Roboto,sans-serif;
	font-weight: 400;
	font-size: 16px;
	text-transform: uppercase;
	color: #fff;
	padding: 8px 32px;
}

.modal-text {

	.input-box {
		height: 41px;
		width: 100%;
		border-radius: 4px;
		border: 1px solid #c7cccf;
		padding: 0 16px;
		box-sizing: border-box;
		line-height: normal;
		font-size: 24px;
	}
}
