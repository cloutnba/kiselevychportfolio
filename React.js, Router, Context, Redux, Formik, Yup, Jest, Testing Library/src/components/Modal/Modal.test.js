import {render} from "@testing-library/react";
import Modal from "./Modal";

describe("Testing Modal Component", () => {
    test("To be in doc", () => {
        const {container} = render(<Modal text="Test Modal" />);

        const modal = container.querySelector('.modal-wrapper');

        expect(modal).toBeInTheDocument;
    });

    test("Check text", () => {
        const {container} = render(<Modal text="Test Modal" />);

        const text = container.querySelector('.modal-text').textContent;

        expect(text).toBe("Test Modal");
    });

    test("Make snapshot", () => {
        const modal = render(<Modal text="Test Modal" />);

        expect(modal).toMatchSnapshot();
    });
})
