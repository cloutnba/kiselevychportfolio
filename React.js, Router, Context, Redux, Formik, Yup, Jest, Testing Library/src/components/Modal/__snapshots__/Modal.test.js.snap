// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal Component Make snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal-wrapper"
      >
        <div
          class="modal"
        >
          <div
            class="modal-box"
          >
            <button
              class="modal-close"
              type="button"
            >
              <svg
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="m9.3 8 6.1-6.1c.4-.4.4-.9 0-1.3s-.9-.4-1.3 0L8 6.7 1.9.6C1.6.3 1 .3.6.6c-.3.4-.3 1 0 1.3L6.7 8 .6 14.1c-.4.4-.3.9 0 1.3l.1.1c.4.3.9.2 1.2-.1L8 9.3l6.1 6.1c.4.4.9.4 1.3 0s.4-.9 0-1.3L9.3 8z"
                />
              </svg>
            </button>
            <div
              class="modal-content"
            >
              <div
                class="modal-text"
              >
                Test Modal
              </div>
            </div>
            <div
              class="modal-footer"
            >
              <div
                class="button-wrapper"
              >
                <button
                  class="btn"
                  type="button"
                >
                  ОК
                </button>
                <button
                  class="btn"
                  type="button"
                >
                  НІ
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal-wrapper"
    >
      <div
        class="modal"
      >
        <div
          class="modal-box"
        >
          <button
            class="modal-close"
            type="button"
          >
            <svg
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="m9.3 8 6.1-6.1c.4-.4.4-.9 0-1.3s-.9-.4-1.3 0L8 6.7 1.9.6C1.6.3 1 .3.6.6c-.3.4-.3 1 0 1.3L6.7 8 .6 14.1c-.4.4-.3.9 0 1.3l.1.1c.4.3.9.2 1.2-.1L8 9.3l6.1 6.1c.4.4.9.4 1.3 0s.4-.9 0-1.3L9.3 8z"
              />
            </svg>
          </button>
          <div
            class="modal-content"
          >
            <div
              class="modal-text"
            >
              Test Modal
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <div
              class="button-wrapper"
            >
              <button
                class="btn"
                type="button"
              >
                ОК
              </button>
              <button
                class="btn"
                type="button"
              >
                НІ
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
